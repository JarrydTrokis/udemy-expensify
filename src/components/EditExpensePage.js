import React, { Component } from 'react'
import { connect } from 'react-redux'
import { editExpense, removeExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'

export class EditExpensePage extends Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }

  onClick = (expense) => {
    this.props.removeExpense({ id: expense.id })
    this.props.history.push('/')
  }

  render () {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button
          onClick={() => { this.onClick(this.props.expense) }}>
          REMOVE
        </button>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: expense => dispatch(removeExpense({ id: expense.id }))
})

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
