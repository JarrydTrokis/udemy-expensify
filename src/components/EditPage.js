import React from 'react'

const EditPage = (props) => {
  console.log(props)
  return (
    <div>
      Editing the expense with the id of {props.match.params.id}
    </div>
  )
}

export default EditPage
