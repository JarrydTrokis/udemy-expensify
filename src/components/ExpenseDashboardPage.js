import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () => (
  <div>
    <br />
    <ExpenseListFilters />
    <br />
    <ExpenseList />
  </div>
)

export default ExpenseDashboardPage
