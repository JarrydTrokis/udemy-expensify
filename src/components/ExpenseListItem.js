import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import { toGBP } from '../helpers/toGBP'

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {toGBP(amount)}
      -
      {moment(createdAt).format('Do MMMM, YYYY')}
    </p>
  </div>
)

export default ExpenseListItem
