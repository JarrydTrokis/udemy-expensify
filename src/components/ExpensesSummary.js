import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import { toGBP } from '../helpers/toGBP'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expensePlural = expenseCount === 1 ? 'expense' : 'expenses'

  return (
    <header>
      <h1>Viewing {expenseCount} {expensePlural} totalling {toGBP(expensesTotal)}</h1>
    </header>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)

  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
