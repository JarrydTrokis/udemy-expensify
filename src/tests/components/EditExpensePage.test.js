import React from 'react'
import { shallow } from 'enzyme'
import { EditExpensePage } from '../../components/EditExpensePage'
import { singleExpense } from '../fixtures/fixtures'

let editExpenseSpy, removeExpenseSpy, historySpy, wrapper

beforeEach(() => {
  removeExpenseSpy = jest.fn()
  editExpenseSpy = jest.fn()
  historySpy = { push: jest.fn() }
  wrapper = shallow(
    <EditExpensePage
      editExpense={editExpenseSpy}
      removeExpense={removeExpenseSpy}
      history={historySpy}
      expense={singleExpense}
    />
  )
})

test('should render EditExpense Page correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle editExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(singleExpense)
  expect(historySpy.push).toHaveBeenLastCalledWith('/')
  expect(editExpenseSpy).toHaveBeenLastCalledWith(singleExpense.id, singleExpense)
})

test('should handle removeExpense', () => {
  wrapper.find('button').simulate('click')
  expect(historySpy.push).toHaveBeenLastCalledWith('/')
  expect(removeExpenseSpy).toHaveBeenLastCalledWith({ id: singleExpense.id })
})
