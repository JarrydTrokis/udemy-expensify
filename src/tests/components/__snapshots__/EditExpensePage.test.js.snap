// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpense Page correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 3000,
        "createdAt": 86400000,
        "description": "food shopping",
        "id": "4",
        "note": "Bought food for the week",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    REMOVE
  </button>
</div>
`;
